### ISSUE TITLE
check_syntax false positive: Not a structure, target, IntegerType on field-symbol assigned via INSERT … INDEX … ASSIGNING

### ISSUE THREAD

**Issue by rubber-duck-53** [abaplint/abaplint#4179]
```abap
REPORT ztest_insert_with_index.

TYPES: BEGIN OF ty_item,
         a   TYPE c LENGTH 1,
         b TYPE c LENGTH 1,
       END OF ty_item.
TYPES: tt_item TYPE STANDARD TABLE OF ty_item WITH EMPTY KEY.

DATA lt_items TYPE tt_item.

INSERT INITIAL LINE INTO lt_items
  INDEX 1
  ASSIGNING FIELD-SYMBOL(<fs_item>).
<fs_item>-a = 'A'.
<fs_item>-b = 'B'.
```
### Expected
No findings —  similar to the case where `INDEX 1` is omitted.

### Actual
```
ztest_insert_with_index.prog.abap[14, 1] - Not a structure, target, IntegerType, - (check_syntax) [E]
ztest_insert_with_index.prog.abap[15, 1] - Not a structure, target, IntegerType, - (check_syntax) [E]
```

**Comment by larshp** [comment]
@rubber-duck-53 try again, v2.120.16 published

**Comment by rubber-duck-53** [comment]
Thanks, works as expected now.
